### PR TITLE
Bug 1813221: jsonnet: add missing resource requests for alertmanager sidecar

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @openshift/openshift-team-monitoring

### CODE REVIEW

File: assets/thanos-querier/deployment.yaml [hunk 1]
```diff
@@ -33,18 +33,14 @@ spec:
             command:
             - sh
             - -c
-            - if [ -x "$(command -v curl)" ]; then curl http://localhost:9090/-/healthy;
-              elif [ -x "$(command -v wget)" ]; then wget --quiet --tries=1 --spider
-              http://localhost:9090/-/healthy; else exit 1; fi
+            - if [ -x "$(command -v curl)" ]; then curl http://localhost:9090/-/healthy; elif [ -x "$(command -v wget)" ]; then wget --quiet --tries=1 --spider http://localhost:9090/-/healthy; else exit 1; fi
```
Comment (s-urbaniak) [inline assets/thanos-querier/deployment.yaml:36]: hmm, these changes are strange, but as far as i see `jb` just undid the newlines? :man_shrugging:

**Comment by s-urbaniak** [comment]
hmm, this is strange, it looks like https://github.com/openshift/prometheus-operator/pull/74 does not show effect here :thinking: 

@pgier any idea? maybe it's just the payload that is not up-to-date yet :thinking:

**Comment by s-urbaniak** [comment]
Strange from what I see in https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/818/pull-ci-openshift-cluster-monitoring-operator-release-4.4-e2e-aws-operator/1275387369729560576/artifacts/e2e-aws-operator/statefulsets.json:
```
$ jq -C '.items[] | select(.metadata.labels.alertmanager == "main")' statefulsets.json | less -R
...
            "name": "config-reloader",
            "resources": {
...
              "requests": {
                "cpu": "1m"
              }
            },

...
```

The cpu request is correctly set in the config-reloader sidecar. Investigating why the schedule failed, maybe events will give a clue.

**Comment by pgier** [comment]
Looks like this is an issue in CMO where the image is getting set by index instead of by name:
https://github.com/openshift/cluster-monitoring-operator/blob/784585a66c36745fae6f46e33bed1ffed1e92384/pkg/manifests/manifests.go#L368

@simonpasquier fixed this in master already (https://github.com/openshift/cluster-monitoring-operator/commit/17319b8ff85456ec51ea576e5b4e86868d03dc3e).  I can work on a backport of that fix for 4.4

File: assets/thanos-querier/deployment.yaml (hunk 1 above)
Comment (paulfantom) [reply assets/thanos-querier/deployment.yaml:36]: I believe this is rather something in gojsontoyaml. We generate json, which doesn't have a concept of line length and then convert it to yaml. Either way I think we can live with it :)

**Pull request merged**